rpm/bundle-rust-downstream

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python3
+
+# --- BEGIN COPYRIGHT BLOCK ---
+# Copyright (C) 2021 Red Hat, Inc.
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ---
+#
+# PYTHON_ARGCOMPLETE_OK
+
+import os
+import sys
+import signal
+import argparse
+import argcomplete
+import shutil
+import toml
+from lib389.cli_base import setup_script_logger
+
+SPECFILE_COMMENT_LINE = 'Bundled cargo crates list'
+START_LINE = f"# {SPECFILE_COMMENT_LINE} - START\n"
+END_LINE = f"# {SPECFILE_COMMENT_LINE} - END\n"
+
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    description="""Add 'Provides: bundled(crate(foo)) = version' to a Fedora based specfile. 
+Additionally, add a helper comment with a comulated License metainfo which is based on Cargo.lock file content.""")
+
+parser.add_argument('-v', '--verbose',
+                    help="Display verbose operation tracing during command execution",
+                    action='store_true', default=False)
+
+parser.add_argument('cargo_lock_file',
+                    help="The path to Cargo.lock file.")
+parser.add_argument('spec_file',
+                    help="The path to spec file that will be modified.")
+parser.add_argument('--backup-specfile',
+                    help="Make a backup of the downstream specfile.",
+                    action='store_true', default=False)
+
+
+# handle a control-c gracefully
+def signal_handler(signal, frame):
+    print('\n\nExiting...')
+    sys.exit(0)
+
+
+def backup_specfile(spec_file):
+    log.info(f"Backing up file {spec_file} to {spec_file}.backup")
+    shutil.copy2(spec_file, f"{spec_file}.backup")
```

Comment:
should we date these files rather than overwriting? If you did this operation twice in a row, you would delete the original backup ...

Reply:
I also thought about that but I was hesitant as it may produce a lot of files that the admin will clean eventually.
But thinking more... I think it's better than losing some info if you run it accidentally twice with a wrong result...

Reply:
You could make a sub directory or a dotfolder for the backups to land in instead. Or you could put them into /tmp? That would keep them available but also makes it a bit neater. 

I guess the other thing, is do we need the backup? We have git for version control ....

Reply:
I agree but maybe it'll be useful for some workflows...
It's just a couple of lines and it's optional. I'll ask Mark what he would like for rpm.mk :)

Reply:
Yeah, it's not the "biggest concern" here, but I get your point that it's useful in some work flows. Given this affects you and @mreynolds389 the most, i'll leave it to your judgement here mate.